### PR TITLE
ntorch.cat/stack will automatically reorder tensors

### PR DESCRIPTION
**Pull request by amspector100**
Currently the following code:

`x = ntorch.zeros(20, 10, names=("a", "b"))`
`y = ntorch.ones(30, 20, names=("b", "a"))`
`torch.cat([x, y], dim = 'b')`

Raises an assertion error because the dimensions of x and y are in different order. This should prevent that from happening. (Having named dimensions means that you shouldn't need to call transpose repeatedly to line up your tensors).

### CODE REVIEW

**Comment by srush** [comment]
Perfect. I'm going to rewrite all the tests to make sure all operations satisfy this property.
